### PR TITLE
Update docs and add new docs for Groups resources

### PR DESCRIPTION
**Pull request by krosn**
- Add 408 responses to certain state changing controllers
- Small description fixes
- Add documentation for new controllers that will be part of groups

### CODE REVIEW

File: Documentation/Identity/Identity_Claim.md [hunk 1]
```diff
@@ -0,0 +1,660 @@
+---
```
Comment (bderveni) [inline Documentation/Identity/Identity_Claim.md:1]: These are the docs that you are adding right?

File: Documentation/Identity/Identity_ClientCredentialClient.md [hunk 2]
```diff
@@ -21,7 +21,7 @@ For HTTP requests and responses, the ClientCredentialClient object has the follo
 Property | Type | Description
  --- | --- | ---
 RoleIds | Guid[] | Gets or sets list of Roles to be assigned to this client. Member role is always required. For security reasons we advise against assigning Admin roles to a client.
-Id | string | Gets or sets client ID for this client. This ID should be a GUID.
+Id | string | Gets or sets secret Id.
```
Comment (bderveni) [inline Documentation/Identity/Identity_ClientCredentialClient.md:24]: This is wrong.

File: Documentation/Identity/Identity_HybridClient.md [hunk 3]
```diff
@@ -26,7 +26,7 @@ RedirectUris | string[] | Gets or sets the allowed URIs to which return tokens o
 PostLogoutRedirectUris | string[] | Gets or sets allowed URIs to redirect to after logout. Wildcards are ignored. URIs must match exactly what you are redirecting to after logout. Maximum 10 for client.
 ClientUri | string | Gets or sets URI to a page with information about client (used on consent screen).
 LogoUri | string | Gets or sets URI to client logo (used on consent screen).
-Id | string | Gets or sets client ID for this client. This ID should be a GUID.
+Id | string | Gets or sets secret Id.
```
Comment (bderveni) [inline Documentation/Identity/Identity_HybridClient.md:29]: This is wrong.

File: Documentation/Identity/Identity_Role.md [hunk 4]
```diff
@@ -0,0 +1,1116 @@
+---
```
Comment (bderveni) [inline Documentation/Identity/Identity_Role.md:1]: Remove this file completely. It should not be here as we do not support this yet/

File: Documentation/Identity/Identity_HybridClient.md (hunk 3 above)
Comment (krosn) [reply Documentation/Identity/Identity_HybridClient.md:29]: Huh, this is the XML on the Id. It looks like it's said that for a while. I'll fix it in this PR manually and make a WI to fix these sumaries

File: Documentation/Identity/Identity_Claim.md [hunk 5]
```diff
@@ -0,0 +1,660 @@
...
+
+### Returns
+
+#### 200
+
+Success.
+
+##### Type:
+
+ `Void`
```
Comment (krosn) [inline Documentation/Identity/Identity_Claim.md:349]: This looks out of place. Is this expected @bderveni

File: Documentation/Identity/Identity_Claim.md (hunk 1 above)
Comment (krosn) [reply Documentation/Identity/Identity_Claim.md:1]: And ClaikmTypeName

**Comment by krosn** [comment]
Yes. Those two changes are correct, the current summaries are wrong. According to our code those are in fact the IDs of the secret, not IDs of the client.

File: Documentation/Identity/Identity_Claim.md [hunk 6]
```diff
@@ -0,0 +1,660 @@
...
+***
+
+## Authentication
+
+All endpoints referenced in this documentation require authenticated access. Authorization header must be set to the access token you retrieve after a successful authentication request.
+
+`Authorization: Bearer <token>`
+
+Requests made without an access token or an invalid/expired token will fail with a 401 Unauthorized response.
+Requests made with an access token which does not have the correct permissions (see security subsection on every endpoint) will fail with a 403 Forbidden.
```
[the remark this answers is not in the document]
Comment (krosn) [reply Documentation/Identity/Identity_Claim.md:46]: Like which above? This appears to be consistent with the other files I have checked

File: Documentation/Identity/Identity_Claim.md [hunk 7]
```diff
@@ -0,0 +1,660 @@
...
+An Identity Provider Claim provides a mechanism to map a Claim
+            from an Identity Provider to one or more Roles.
+
+## Properties
+
+For HTTP requests and responses, the Claim object has the following properties and JSON-serialized body: 
+
+Property | Type | Description
+ --- | --- | ---
+Issuer | string | Gets or sets the issuer of the claim.
```
[the remark this answers is not in the document]
Comment (krosn) [reply Documentation/Identity/Identity_Claim.md:16]: Changes

**Pull request merged**